src/vm/sampleprofiler.h

```diff
@@ -43,6 +43,22 @@ class SampleProfiler
         // Profiling thread proc.  Invoked on a new thread when profiling is enabled.
         static DWORD WINAPI ThreadProc(void *args);
 
+        // Calls either PAL_nanosleep or ClrSleepEx depending on platform
+        // Note: Although we specify the time in ns, that is no indication
+        // of the actually accuracy with whihc we will return from sleep
+        // In reality Unix will have a minimum granularity of ~10ms
+        // and Windows has a default granularity of ~16ms, but can be
+        // adjusted to as low as ~1ms
+        // Even this however is not gaurenteed. If the system is under load
+        // the sampling thread may be delayed up to hundreds of ms due to
+        // scheduling priority. There is no way to prevent this from user threads
+        // Additionally we may get lucky and there will be an open CPU to run
+        // and under light load the timings will achieve great accuracy!
+        static void PlatformSleep(long nanoseconds);
+
+        static void BeginTimePeriod();
```

Comment:
I think that Begin/End TimePeriod functions should have clearer names for what they are actually doing with respect to the profiler.  They are setting and re-setting the scheduling granularity.  They should probably be called something like SetSchedulerGranularity and ResetSchedulerGranularity.  Thoughts?